core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -851,13 +853,41 @@ class UserService(
     billingAccount: Option[RawlsBillingAccountName]
   ): Future[Option[RawlsBillingProjectResponse]] = for {
     project <- updateBillingAccountInDatabase(projectName, billingAccount)
+    _ <- project
+      .collect { p =>
+        p.billingProfileId.collect { pf =>
+          updateBillingAccountInBillingProfile(pf, billingAccount)
+        }
+      }
+      .flatten
+      .sequence
     projectRoles <- samDAO
       .listUserRolesForResource(SamResourceTypeNames.billingProject, projectName.value, ctx)
       .map(resourceRoles => samRolesToProjectRoles(resourceRoles))
   } yield project.flatMap { p =>
     if (projectRoles.nonEmpty) Some(RawlsBillingProjectResponse(projectRoles, p)) else None
   }
 
+  private def updateBillingAccountInBillingProfile(billingProfileId: String,
+                                                   billingAccount: Option[RawlsBillingAccountName]
+  ): Future[Unit] =
+    (billingAccount match {
+      case Some(newBillingAccount) =>
+        billingProfileManagerDAO
+          .updateBillingProfile(UUID.fromString(billingProfileId), newBillingAccount, ctx)
+          .flatMap(_ => Future.unit)
+      case None =>
+        billingProfileManagerDAO.removeBillingAccountFromBillingProfile(UUID.fromString(billingProfileId), ctx)
+    }).recover {
+      // Until BPM is the system of record for Terra billing information, Rawls will not throw an exception if BPM fails to update
+      case e: ApiException =>
```

Comment:
Nit: Should this cast a wider net, like Exception? I am thinking about ProcessingExceptions, SocketExceptions etc. that can crop up that are not in the ApiException hierarchy.

Reply:
Yeah, that's a good call. I was mostly concerned about preventing bugs in BPM from causing issues, but this should really cover any issues related to the call to BPM